Muehle_Logic.ipynb

```diff
@@ -201,6 +201,83 @@
+    "                    print('Das ist nicht Dein Stein. Bitte probiere es erneut!')\n",
+    "\n",
+    "            elif status.move_or_jump == True: # -> Zielfeld für Stein wählen\n",
+    "                old_ring, old_cell = status.selected_dot_old\n",
+    "                new_ring, new_cell = status.selected_dot_new\n",
+    "                possible_positions = next_positions(status.pieces, status.current_player, old_ring, old_cell) # ausgehend vom ausgewählten Stein, erlaubte Zielfelder berechnen\n",
+    "                if (new_ring, new_cell) in possible_positions: # prüfen ob das ausgewählte Feld ein erlaubtes Zielfeld ist\n",
+    "                    status.pieces[1][new_ring][new_cell] = status.current_player # das Zielfeld als das eigene markieren\n",
+    "                    status.pieces[1][old_ring][old_cell] = 0 # das ursprüngliche Feld ist jetzt leer\n",
+    "                    status.move_or_jump = False\n",
+    "                    update_board(status)\n",
+    "                else:\n",
+    "                    raise InvalidMoveException()\n",
+    "                    print('Du kannst Deinen Stein nicht auf dieses Feld setzen. Bitte probiere es erneut!')\n",
+    "        else:\n",
+    "            raise InvalidMoveException()\n",
+    "            print('Du befindest Dich anscheinend in keiner gültigen Spielphase. Irgendwas ist schief gegangen... Sorry!')\n",
+    "        \n",
+    "        status.number_stones_to_remove = handle_mills(status)\n",
+    "        if status.number_stones_to_remove > 0:\n",
+    "            show_possible_clickable_positions(get_opponent_beatable_stones(status.current_player), colour = COLOUR_OPPONENT)\n",
+    "    else:\n",
+    "        selected_ring, selected_cell = status.selected_dot_new\n",
+    "        beatable_stones = get_opponent_beatable_stones(status.current_player)\n",
+    "        if (selected_ring, selected_cell) in beatable_stones:\n",
+    "            status.pieces[1][selected_ring][selected_cell] = 0\n",
+    "            status.number_stones_to_remove = status.number_stones_to_remove - 1\n",
+    "            update_board(status)\n",
+    "        else:\n",
+    "            raise InvalidMoveException()\n",
+    "            print('Du kannst diesen Stein nicht entfernen. Bitte probiere es erneut!')\n",
+    "        if status.number_stones_to_remove > 0:\n",
+    "            show_possible_clickable_positions(get_opponent_beatable_stones(status.current_player), colour = COLOUR_OPPONENT)\n",
+    "    \n",
+    "    if status.move_or_jump == False and status.number_stones_to_remove == 0:\n",
+    "        status.current_player = opponent(status.current_player) # Spielerwechsel, wenn nicht noch eine Aktion ausgeführt werden muss\n",
+    "        \n",
+    "    if finish(status.pieces) != 0: # Prüfen, ob das Spiel zu Ende ist\n",
+    "        status.game_over = True\n",
+    "        winner_popup(finish(status.pieces))"
+   ]
+  },
```

Comment:
-Ich konnte hier zwar logisch keinen Fehler erkennen, aber ich konnte trotzdem teilweise seltsames Verhalten feststellen (siehe Discord)
-Auch hier stellt sich die Frage, ob wir die Texte in Konstanten auslagern, oder so lassen wollen.

Reply:
Siehe https://github.com/luisafuss/muehle-studienarbeit/issues/38, https://github.com/luisafuss/muehle-studienarbeit/issues/39, https://github.com/luisafuss/muehle-studienarbeit/issues/40